### DISCUSSION TITLE
GetAtt, Text or Array?

### DISCUSSION THREAD

**Discussion by denizdogan** [jcouyang/dhall-aws-cloudformation#15] [Q&A]
I'm having a hard time understanding `GetAtt`. The definition is currently:

```dhall
, GetAtt =
    λ(x : Text) → JSON.object (toMap { `Fn::GetAtt` = JSON.string x })
```

But the value should be an array, according to [AWS documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-getatt.html):

```json
{ "Fn::GetAtt" : [ "logicalNameOfResource", "attributeName" ] }
```

Is this a bug in the Dhall library or am I missing something here?

**Comment by jcouyang** [comment]
`GetAtt` implemented the short form `!GetAtt logicalNameOfResource.attributeName`
```dhall
let exampleGetAtt =
        assert
      :   JSON.render (toJSON (GetAtt "myELB.DNSName"))
        ≡ ''
          { "Fn::GetAtt": "myELB.DNSName" }
          ''
```

actually I'm using GetAtt in another way, which is not documented but I think it will be more typefuly correct

For all resources there is an `GetAtt` record, containing all possible attributes for the specific type, e.g.
```dhall
let S3 = https://raw.githubusercontent.com/jcouyang/dhall-aws-cloudformation/0.9.69/cloudformation/AWS::S3::Bucket.dhall
in S3.GetAttr.DomainName "s3LogicName"
```
